utils/environment.go

```diff
@@ -68,13 +68,13 @@ type envEnvelope struct {
 	Languages  LanguageList `json:"languages"`
 }
 
-func ReadEnvironment(data *json.RawMessage) (*environment, error) {
+func ReadEnvironment(data json.RawMessage) (*environment, error) {
```

Comment:
Yep, change is good, just making sure you knew that slices are definitely not pass by reference as this is a common go error. (you can pass a map that way and it is ok to change it, just not for slices)